erpnext/selling/sales_common.js

```diff
@@ -151,9 +151,27 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
 	order_type: function() {
 		if (this.frm.doc.order_type) {
 			this.set_and_update_excise_tax();
+			this.frm.trigger("set_expense_discount");
 		}
 	},
 
+	set_expense_discount: function(frm) {
+		let me = this;
+		let percentage_discount = 0;
+
+		if (me.frm.doc.order_type === "Marketing") {
+			me.frm.set_value("apply_discount_on", "Grand Total");
+			percentage_discount = 100;
+		}
+
+		me.frm.set_value("additional_discount_percentage", percentage_discount);
```

Comment:
Do this on the server side as well (on save), I was able to start an invoice on "Marketing", but it didn't apply the discount because there was no change in the Order Type

Reply:
@Alchez  Server side check added